### PR TITLE
feat: TDE for PG clusters

### PR DESCRIPTION
**Pull request by wai-wong-edb**

### CODE REVIEW

**Review by vishalsawale9 — APPROVED** [review]
Looks good to me. It may be worth adding some code comments in the diff.

File: pkg/provider/data_source_cluster.go [hunk 1]
```diff
@@ -359,6 +359,29 @@ func (c *clusterDataSource) Schema(ctx context.Context, req datasource.SchemaReq
...
+						Computed:            true,
+					},
+					"status": schema.StringAttribute{
+						MarkdownDescription: "Status.",
+						Computed:            true,
+					},
+				},
+			},
+			"pg_identity": schema.StringAttribute{
+				MarkdownDescription: "PG Identity.",
```
Comment (edbamolvyavahare) [inline pkg/provider/data_source_cluster.go:381]: PG Identity required to grant key permissions to activate the cluster.

File: pkg/provider/resource_cluster.go [hunk 2]
```diff
@@ -472,6 +481,38 @@ func (c *clusterResource) Schema(ctx context.Context, req resource.SchemaRequest
...
+						},
+					},
+					"status": schema.StringAttribute{
+						MarkdownDescription: "Status.",
+						Computed:            true,
+					},
+				},
+			},
+			"pg_identity": schema.StringAttribute{
+				MarkdownDescription: "PG Identity.",
```
Comment (edbamolvyavahare) [inline pkg/provider/resource_cluster.go:513]: PG Identity required to grant key permissions to activate the cluster.

File: pkg/provider/resource_cluster.go [hunk 3]
```diff
@@ -742,6 +783,7 @@ func readCluster(ctx context.Context, client *api.ClusterClient, tfClusterResour
 	tfClusterResource.FarawayReplicaIds = StringSliceToSet(responseCluster.FarawayReplicaIds)
 	tfClusterResource.PrivateNetworking = types.BoolPointerValue(responseCluster.PrivateNetworking)
 	tfClusterResource.SuperuserAccess = types.BoolPointerValue(responseCluster.SuperuserAccess)
+	tfClusterResource.PgIdentity = types.StringValue(*responseCluster.PgIdentity)
```
Comment (edbamolvyavahare) [inline pkg/provider/resource_cluster.go:786]: this should also be defined under 
`if responseCluster.EncryptionKey != nil {}` block

File: examples/resources/biganimal_cluster/single_node/aws/resource.tf [hunk 4]
```diff
@@ -104,6 +104,10 @@ resource "biganimal_cluster" "single_node_cluster" {
   # pe_allowed_principal_ids = [
   #   <example_value> # ex: 123456789012
   # ]
+
+  # transparent_data_encryption = {
+  #   key_id = <example_value>
+  # }
```
[the remark this answers is not in the document]
Comment (wai-wong-edb) [reply examples/resources/biganimal_cluster/single_node/aws/resource.tf:110]: needed

**Review by vishalsawale9 — DISMISSED** [review]
Looks good to me.

File: pkg/provider/helpers.go [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+func TdeActionInfo(provider string) string {
+	switch provider {
+	case "aws", "bah:aws":
+		return fmt.Sprintf(constants.TDE_KEY_AWS_ACTION, provider)
+	case "azure", "bah:azure":
+		return fmt.Sprintf(constants.TDE_KEY_AZURE_ACTION, provider)
+	case "gcp", "bah:gcp":
+		return fmt.Sprintf(constants.TDE_KEY_GCP_ACTION, provider)
+	default:
+		return fmt.Sprintf(constants.TDE_KEY_ACTION_UNKNOWN_PROVIDER_ERROR, provider)
```
Comment (edbamolvyavahare) [inline pkg/provider/helpers.go:18]: - constants.TDE_KEY_AWS_ACTION message should be displayed for both BAH and BYOA providers
- exception: we do not support TDE for bah:azure yet
```suggestion
	switch provider {
	case "aws", "bah:aws":
		return fmt.Sprintf(constants.TDE_KEY_AWS_ACTION, provider)
	case "azure", "bah:azure":
		return fmt.Sprintf(constants.TDE_KEY_AZURE_ACTION, provider)
	case "gcp", "bah:gcp":
		return fmt.Sprintf(constants.TDE_KEY_GCP_ACTION, provider)
	default:
		return fmt.Sprintf(constants.TDE_KEY_ACTION_UNKNOWN_PROVIDER_ERROR, provider)
```

**Review by edbamolvyavahare — APPROVED** [review]
LGTM

**Pull request merged**